internal/service/rds/cluster_fw.go

```diff
@@ -0,0 +1,2169 @@
+							PlanModifiers: []planmodifier.Int64{
+								fwint64planmodifier.DefaultValue(300),
+							},
+							Validators: []validator.Int64{
+								int64validator.Between(300, 86400),
+							},
+						},
+						"timeout_action": schema.StringAttribute{
+							Optional: true,
+							Computed: true,
+							PlanModifiers: []planmodifier.String{
+								fwstringplanmodifier.DefaultValue(TimeoutActionRollbackCapacityChange),
+							},
+							Validators: []validator.String{
+								stringvalidator.OneOf(TimeoutAction_Values()...),
+							},
+						},
+					},
+				},
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+				},
+			},
+			"serverlessv2_scaling_configuration": schema.ListNestedBlock{
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"max_capacity": schema.Float64Attribute{
+							Required: true,
+							Validators: []validator.Float64{
+								float64validator.Between(0.5, 128),
+							},
+						},
+						"min_capacity": schema.Float64Attribute{
+							Required: true,
+							Validators: []validator.Float64{
+								float64validator.Between(0.5, 128),
+							},
+						},
+					},
+				},
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+				},
+			},
+		},
+	}
+
+	if s.Blocks == nil {
+		s.Blocks = make(map[string]schema.Block)
+	}
+	s.Blocks["timeouts"] = timeouts.Block(ctx, timeouts.Opts{
+		Create: true,
+		Update: true,
+		Delete: true,
+	})
+
+	response.Schema = s
+}
+
+// Create is called when the provider must create a new resource.
+// Config and planned state values should be read from the CreateRequest and new state values set on the CreateResponse.
+func (r *resourceCluster) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	var data resourceClusterData
+	conn := r.Meta().RDSConn()
+
+	response.Diagnostics.Append(request.Plan.Get(ctx, &data)...)
+
+	defaultTagsConfig := r.Meta().DefaultTagsConfig
+	ignoreTagsConfig := r.Meta().IgnoreTagsConfig
+	tags := defaultTagsConfig.MergeTags(tftags.New(data.Tags))
+	data.TagsAll = flex.FlattenFrameworkStringValueMapLegacy(ctx, tags.IgnoreAWS().IgnoreConfig(ignoreTagsConfig).Map())
+
+	if response.Diagnostics.HasError() {
+		return
+	}
+	createTimeout := r.CreateTimeout(ctx, data.Timeouts)
+
+	var requiresModifyDbCluster bool
+	modifyDbClusterInput := &rds.ModifyDBClusterInput{
+		ApplyImmediately: aws.Bool(true),
+	}
+
+	var identifier string
+	if !data.ClusterIdentifier.IsUnknown() && !data.ClusterIdentifier.IsNull() {
+		identifier = data.ClusterIdentifier.ValueString()
+	} else if !data.ClusterIdentifierPrefix.IsUnknown() && !data.ClusterIdentifierPrefix.IsNull() {
+		identifier = sdkresource.PrefixedUniqueId(data.ClusterIdentifierPrefix.ValueString())
+	} else {
+		identifier = sdkresource.PrefixedUniqueId("tf-")
+	}
+
+	var output *rds.DBCluster
+	if !data.SnapshotIdentifier.IsUnknown() && !data.SnapshotIdentifier.IsNull() {
```

Comment:
Would it make sense to break out the bodies of the conditionals into functions instead of leaving them inline?